server/lib/chat/chat-service.ts

```diff
@@ -176,25 +201,34 @@ export default class ChatService {
     let channel: ChannelInfo | undefined
     let userChannelEntry: UserChannelEntry | undefined
     let message: ChatMessage
+    let error: ChatServiceError | undefined
     do {
       attempts += 1
       try {
         await transact(async client => {
           channel = await findChannelByName(channelName, client)
           if (channel) {
             if (this.state.users.has(userId) && this.state.users.get(userId)!.has(channel.id)) {
-              throw new ChatServiceError(
+              error = new ChatServiceError(
                 ChatServiceErrorCode.AlreadyJoined,
                 'Already in this channel',
               )
+              return
             }
 
             const isBanned = await isUserBannedFromChannel(channel.id, userId, client)
             if (isBanned) {
-              throw new ChatServiceError(
+              error = new ChatServiceError(
+                ChatServiceErrorCode.UserBanned,
+                'This user has been banned from this chat channel',
```

Comment:
These messages are potentially user-visible and should be phrased as such, they are not for our log purposes